R/cdfPlot.R

```diff
@@ -146,110 +252,97 @@ cdfPlot <- function(contrastDF,
     # Combo PLOT: full data inset, most significant data in main plot
     # Rank by p-value
     contrastDF <- contrastDF %>%
-        dplyr::arrange(!!sym(y))
+        dplyr::arrange(!!sym(pvalCol))
     contrastDF$Rank <- c(1:nrow(contrastDF))
 
-    # Let's plot the p-value subsets
-    Sig <- contrastDF[[y]] <= pThreshold
-    NotSig <- contrastDF[[y]] > pThreshold
-
-    # Create group factor column in contrastDF
+    # Let"s plot the p-value subsets
     contrastDF$group <- NA
-    contrastDF$group[Sig] <- "Significant"
-    contrastDF$group[NotSig] <- "Not Significant"
-
-    contrastDF$group <- contrastDF$group %>%
-        factor(levels = groupNames)
-
-    # Set an order field to force plotting of 'not significant' first
     contrastDF$order <- NA
-    contrastDF$order[NotSig] <- 1
-    contrastDF$order[Sig] <- 2
+    contrastDF <- contrastDF %>%
+        dplyr::mutate(group = dplyr::case_when(!!rlang::sym(pvalCol) <= pThreshold ~ "Significant",
+                                               !!rlang::sym(pvalCol) > pThreshold ~ "Not Significant"),
+                      group = factor(group,
+                                     levels = c("Significant", "Not Significant")),
+                      order = dplyr::case_when(group == "Not Significant" ~ 1,
+                                               group == "Significant" ~ 2))
 
-    # Rows to include in the zoomed in plot
-    subsetRows <- sum(contrastDF[[y]] <= pvalMax)
-    contrastDFsubset <- contrastDF[1:subsetRows,]
+    contrastDF_subset <- contrastDF %>%
+        dplyr::filter(!!rlang::sym(pvalCol) <= pvalMax)
+    cdfMain <- NULL
+    cdfInset <- NULL
 
-    if (plotType == "canvasXpress") {
+    if (plotType == "canvasxpress") {
         ## Create the canvasXpress cx.data and var.annot
         # Main plot
-        cx.data <- data.frame(a = contrastDF[colnames(contrastDF) == x],
-                              b = contrastDF[colnames(contrastDF) == y])
+        cx.data <- contrastDF %>% dplyr::select(!!x,!!y)
         colnames(cx.data) <- c(x, y)
-        var.annot <- data.frame(Group = contrastDF$group)
-        rownames(var.annot) <- rownames(cx.data)
+        var.annot <- contrastDF %>% dplyr::select(group)
```

Comment:
add select to importFrom tag as well

Reply:
already added. are you not able to see the latest code ? Can you check once more
'?

Reply:
yes. it looks like you are looking at a older version for some reason.. try again now.